db/crud.go

```diff
@@ -126,7 +134,15 @@ func (db *DatabaseContext) GetDocSyncData(docid string) (syncData, error) {
 		// If existing doc wasn't an SG Write, import the doc.
 		if !doc.IsSGWrite() {
 			var importErr error
-			doc, importErr = db.OnDemandImportForGet(docid, rawDoc, rawXattr, cas)
+
+			// Get the doc doc expiry
+			gocbBucket := db.Bucket.(*base.CouchbaseBucketGoCB)
+			expiry, getExpiryErr := gocbBucket.GetExpiry(key)
+			if getExpiryErr != nil {
+				return emptySyncData, getExpiryErr
+			}
+
```

Comment:
We call OnDemandImportForGet in a couple of places, and I don't think already have the expiry value in either of them.  Can we move the expiry retrieval inside OnDemandImportForGet?

Reply:
From chat:

> initially my thought was "we need to retrieve the Cas that's associated with the expiry when we do GetExpiry, and make sure it matches doc.Cas".
> but thinking further I don't think that's actually required.  If the doc was updated in between us getting existingDoc, and getting the expiry, the subsequent write is going to fail anyway (due to CAS)

So no action needed on this